client/src/helpers/AuthSubmitHelper.ts

```diff
@@ -0,0 +1,32 @@
+import { login, refreshToken, registration } from '../api/UserApi';
+
+import { LoginFormType } from '../screens/LoginScreen';
+import { RegistrationFormType } from '../screens/RegistrationScreen';
+import { StatusEnum } from '../store/types';
+import { userStore } from '../store/UserStore';
+
+import { displayErrorToast } from './ToastDisplayHelper';
+
+export const authSubmitHelper = async (values?: LoginFormType | RegistrationFormType) => {
+    try {
+        let user;
+        userStore.setStatus(StatusEnum.loading);
+
+        if (!values) {
+            user = await refreshToken();
+        } else if ((values as RegistrationFormType).phoneNumber) {
+            user = await registration(values as RegistrationFormType);
+        } else {
+            user = await login(values as LoginFormType);
+        }
```

Comment:
I don't think that it's good practice that this function depends on some fields of the argument. Can you think of a better approach?

Reply:
I am talking about this if else